First attempt at fixing #3132.

djl opened the pull request:
Add a new `no_clobber` config option that contains a list of "miscellaneous" metadata fields not to be overwritten by empty values.

I'm not entirely happy with the `no_clobber` name. Suggestions welcome.

sampsyo commented on beets/autotag/__init__.py:
```diff
@@ -154,9 +154,14 @@ def apply_metadata(album_info, mapping):
                       'data_source',):
+            # Don't overwrite fields with empty values unless the
+            # field is explicitly allowed to be overwritten
+            clobber = field not in config['no_clobber'].get()
```
A type assertion here might be useful: `.as_str_seq()` instead of `.get()`.

djl replied:
Good catch. Fixed.

sampsyo commented:
Hmm... I’m not sure what the best course of action is. Maybe it would be simplest to invert the sense of the option (overwrite_null?), leave it empty by default, and possibly leave it undocumented for now because of its hard-to-explain limitations?

djl commented:
Updated the code to handle the track-level fields and inverted the logic as asked. Also removed the documentation and changelog entries for now.

And it looks like Travis is having a bad day.

djl commented:
All green :+1:

sampsyo commented:
Awesome! This looks like it will be useful to build on, _and_ it made the code for that function simpler. Thanks for your help with this!! :sparkles:
